src/components/ContextMenu/index.js

```diff
@@ -0,0 +1,109 @@
+import React, { useState, useRef, useEffect, useLayoutEffect } from 'react';
+import styled from 'styled-components';
+import ContextMenuItem from './ContextMenuItem';
+
+function ContextMenu({ items, mousePos, parentRef, isToLeft, displayFocus }) {
+  const [isVisible, setIsVisible] = useState(false);
+  const [openToLeft, setOpenToLeft] = useState(false);
+  const [menuRef, setRefPos] = useState(null);
+  const [menuPos, setMenuPos] = useState({
+    top: '',
+    left: '',
+  });
+  const elementRef = useRef(null);
+
+  useLayoutEffect(() => {
+    if (displayFocus || parentRef)
+      setRefPos(elementRef.current.getBoundingClientRect());
+  }, []);
```

Comment:
⚠️ This and the other dependencies array are missing some deps - Please make sure ESlint is enabled in your editor and that you see warnings & errors and work through them.

Reply:
Fixed.